java-tools/OsmAndServer/src/main/java/net/osmand/server/controllers/pub/GeotiffTileController.java

```diff
@@ -80,6 +81,35 @@ public ResponseEntity<?> getTile(@PathVariable String type, @PathVariable int z,
 		return ResponseEntity.ok(new ByteArrayResource(baos.toByteArray()));
 	}
 
+	@Scheduled(fixedRate = CLEANUP_INTERVAL_MILLIS)
+	public void cleanUpCache() {
+		File cacheDir = new File(config.heightmapLocation);
+		if (!cacheDir.exists() || !cacheDir.isDirectory()) {
+			return; // Nothing to clean up
+		}
+		cleanUpDirectory(cacheDir);
+	}
+
+	private void cleanUpDirectory(File dir) {
+		long now = System.currentTimeMillis();
+		File[] files = dir.listFiles();
+
+		if (files != null) {
+			for (File file : files) {
+				if (file.isDirectory()) {
+					cleanUpDirectory(file);
+				} else if (file.isFile() && now - file.lastModified() >= SEVEN_DAYS_IN_MILLIS) {
+					try {
+						Path filePath = file.toPath();
+						Files.delete(filePath);
+					} catch (IOException e) {
+						LOGGER.warn("Failed to delete file: " + file.getAbsolutePath(), e);
+					}
+				}
+			}
+		}
+	}
```

Comment:
Please validate filename (by the mask at least) to prevent accidental "system cleanup" if heightmapLocation is set as "/"